frontend/src/pages/notebookController/screens/server/SizeSelectField.tsx

```diff
@@ -0,0 +1,55 @@
+import * as React from 'react';
+import { FormGroup, Select, SelectOption } from '@patternfly/react-core';
+import AppContext from '../../../../app/AppContext';
+
+type SizeSelectFieldProps = {
+  value: string;
+  setValue: (newValue: string) => void;
+};
+
+const SizeSelectField: React.FC<SizeSelectFieldProps> = ({ value, setValue }) => {
+  const [sizeDropdownOpen, setSizeDropdownOpen] = React.useState<boolean>(false);
+  const { dashboardConfig } = React.useContext(AppContext);
+  console.log(value);
+
+  const sizeOptions = React.useMemo(() => {
```

Comment:
Don't memo.

---
I'm going to try and explain a few things to help you understand why not to memo in this case. Hopefully this is helpful.

When creating a list of React components, they are in themselves "memoized" by React. That's what the `key` prop is doing. 

When React diffs the DOM on each render to see if it needs to mutate it, it will check static items have the same props they had before -- any divs, form elements, the select that PF mounts, etc... and it will get down to your `sizeOptions`, the things you are memoizing, and each one will have a key that says "I'm Small" or "I'm Large" and React will see that your items are still in `Small, Medium, Large` order, and infer based on the keys that nothing has changed order. It compares the props each has -- "yup, those are the same too". React will do no DOM changes on each render as none of the items have changed in anyway.

What I think you are trying to do, is save yourself cycles on recomputing the React objects -- but we don't really care about that. It's like saving yourself from `1 + 2` on each render -- computers do this incredibly fast. You're probably taking up more memory by using memo than saving on processing power computing the items each render.

The things we need to worry about in the web world are DOM manipulations and API calls. The former is solved FOR US by React -- that's why we use it, no messy jQuery that we need to do it ourselves. The latter is solved by putting loading spinners up when we make API calls and using `useEffect` to wrap those in another thread -- allows an app to be reactive all the time.

Try not to repeat logic that React is doing for us... with each extra line you write, you add more for us to maintain and read around to get to the actual items we need. Go simple first. If we need memos, we can add them later. If you need object references to be pure referentially (ie those passed to a hook dependency array) then feel free to memo. But don't memo unless you found a reason to.